crates/bevy_ecs/src/storage/table.rs

```diff
@@ -327,12 +327,17 @@ impl Table {
     pub fn reserve(&mut self, amount: usize) {
         let available_space = self.capacity - self.len();
         if available_space < amount {
-            let reserve_amount = (amount - available_space).max(self.grow_amount);
+            let min_capacity = self.len() + amount;
+            // normally we would check if min_capacity is 0 for the below calculation, but amount > available_space and
+            // available_space > 0, so min_capacity > 1
+            let new_capacity =
+                ((min_capacity + self.grow_amount - 1) / self.grow_amount) * self.grow_amount;
```

Comment:
I think the `self.grow_amount` cancels itself in the div/mul ?

Reply:
No, it rounds down.